pkg/core/engine/main.go

```diff
@@ -504,9 +504,9 @@ func (e *Engine) LoadAutoDiscovery(defaultEnabled bool) error {
 
 			manifest.PipelineID = pipelineID
 
-			if sc != nil {
-				manifest.SCMs = make(map[string]scm.Config)
-				manifest.SCMs[p.Config.Spec.AutoDiscovery.ScmId] = *sc
+			manifest.SCMs = make(map[string]scm.Config)
+			for scmId, sc := range p.SCMs {
+				manifest.SCMs[scmId] = *sc.Config
```

Comment:
@olblak This is the biggest change in this pr in term of complete workflow, as instead of only adding the initial manifest scm to the generate manifests, I forward all scm spec to the generated manifests

Reply:
Interesting, I am curious what's the motivation?

Reply:
The recomended way to parse cargo registries is to pull git repositories containing the index. (In https://github.com/updatecli/updatecli/pull/1081 I've implemented API support because the official cargo registry is big and the clone operation takes about 1 minute).

This means that if we want to run autodiscovery in a git repo (leveraging the current scmid process of autodiscovery), I cannot use a cargo registry in a git scm without this changes

Reply:
Thanks, the change is legit and I can't think of a limitation excepted being sure that there is no collision between the scmid used by the registry versus the one specified in the autodiscovery manifest